apps/server/src/modules/tool/service/tool-validation.service.spec.ts

```diff
@@ -72,27 +75,39 @@ describe('ToolValidation', () => {
 			});
 		});
 
-		describe('hasDuplicateAttributes', () => {
-			it('should not find duplicate custom parameters', async () => {
-				const externalToolDO: ExternalToolDO = externalToolDOFactory.withCustomParameters(2).build();
-
-				const result: Promise<void> = service.validateCreate(externalToolDO);
-
-				await expect(result).resolves.not.toThrow();
+		describe('hasDuplicateAttributes is called', () => {
+			describe('when duplicates are given', () => {
+				it('should throw when external tool', async () => {
+					const externalToolDO: ExternalToolDO = externalToolDOFactory.build({
+						parameters: [
+							customParameterDOFactory.build({ name: 'param1CaseSensitive' }),
```

Comment:
I would prefer the case sensitivity to be tested in a case seperate from the "standard case"